### PR TITLE
OCPBUGS-5692: 4.10 - remove goutils from dependency tree

### PR DESCRIPTION
**Pull request by cheesesashimi**
**- What I did**

Analyzed the MCO to understand where goutils was being introduced as a dependency. I discovered that it was introduced via the sprig template helper library. We only make use of two functions within that library (which do not depend on goutils). This enabled me to do the following:
- Copy those funcs we use into our codebase.
- Consolidate the use of those funcs into a single place. Further consolidation is possible, but outside the scope of this PR.
- Wrote smoke test to validate that all templated manifests can be rendered and then decoded into the appropriate structs.
- Removed sprig, goutils, and its dependent libraries from our vendor folder and go.mod / go.sum files.

Note to reviewer: I put the dependency removal in a commit by itself for easier review. This is manual cherry-pick of https://github.com/openshift/machine-config-operator/pull/3480.

**- How to verify it**

Run the unit test suite

**- Description for the changelog**
Remove goutils from our dependency tree

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -84,13 +84,11 @@ github.com/InVisionApp/go-logger v1.0.1/go.mod h1:+cGTDSn+P8105aZkeOfIhdd7vFO5X1
 github.com/JeffAshton/win_pdh v0.0.0-20161109143554-76bb4ee9f0ab/go.mod h1:3VYc5hodBMJ5+l/7J4xAyMeuM2PNuepvHlGs8yilUCA=
 github.com/MakeNowJust/heredoc v0.0.0-20170808103936-bb23615498cd h1:sjQovDkwrZp8u+gxLtPgKGjk5hCxuy2hrRejBTA9xFU=
 github.com/MakeNowJust/heredoc v0.0.0-20170808103936-bb23615498cd/go.mod h1:64YHyfSL2R96J44Nlwm39UHepQbyR5q10x7iYa1ks2E=
-github.com/Masterminds/goutils v1.1.0 h1:zukEsf/1JZwCMgHiK3GZftabmxiCw4apj3a28RPBiVg=
 github.com/Masterminds/goutils v1.1.0/go.mod h1:8cTjp+g8YejhMuvIA5y2vz3BpJxksy863GQaJW2MFNU=
```
Comment (sergiordlr) [inline go.sum:87]: Here we have a reference to Masterminds/goutils in go.sum

Should it be removed? In similar PRs this reference was removed too. For example https://github.com/openshift/machine-config-operator/pull/3496/files

There is another reference in line 151:
github.com/aokoli/goutils v1.0.1/go.mod h1:SijmP0QR8LtwsmDs8Yii5Z/S4trXFGFC2oO5g9DP+DQ=

Should it be removed too?

Comment (cheesesashimi) [reply go.sum]: Nice catch! It looks like golangci-lint has an indirect dependency on goutils too. I'll push another commit to take care of that by updating it to v1.49.0, which is the version used in 4.11 and 4.12.

**Comment by cheesesashimi** [comment]
I've undone the e2e timeout increases so that https://github.com/openshift/machine-config-operator/pull/3517 can incorporate them instead. That means https://github.com/openshift/machine-config-operator/pull/3517 will have to land before this PR does.

**Comment by yuqi-zhang** [comment]
/override e2e-agnostic-upgrade 

This has passed on the current commit but is very flaky. Should be safe to override given context of this PR

**Comment by yuqi-zhang** [comment]
sorry wait I think I need the whole context

/override ci/prow/e2e-agnostic-upgrade

**Pull request merged**